### PR TITLE
isbn verifier

### LINKED ISSUE #192: Add Practice Exercise: `isbn-verifier`
Problem specification:
https://github.com/exercism/problem-specifications/tree/main/exercises/isbn-verifier
    
TODO:
    
- generate the exercise with `./bin/fetch-configlet && ./bin/configlet create --practice-exercise isbn-verifier`
- check if any of the existing tracks (examples below) implement the exercise and you can reuse the implementation:
    - Go: https://github.com/exercism/go/tree/main/exercises/practice
    - Rust: https://github.com/exercism/rust/tree/main/exercises/practice
    - Python: https://github.com/exercism/python/tree/main/exercises/practice
- use exercise's _canonical-data.json_ to see which test cases to implement:
    https://github.com/exercism/problem-specifications/blob/main/exercises/isbn-verifier/canonical-data.json

### PR DESCRIPTION
**Pull request by Falilah**
Closes #192

### CODE REVIEW

File: exercises/practice/isbn-verifier/src/lib.cairo [hunk 1]
```diff
@@ -0,0 +1,3 @@
+pub fn is_valid_isbn(isbn: ByteArray) -> bool {
```
Comment (0xNeshi) [inline exercises/practice/isbn-verifier/src/lib.cairo:1]:
```suggestion
pub fn is_valid(isbn: ByteArray) -> bool {
```

nit: rename to match canonical data

File: config.json [hunk 2]
```diff
@@ -851,6 +851,14 @@
...
         "prerequisites": [],
         "difficulty": 6
+      },
+      {
+        "slug": "isbn-verifier",
+        "name": "ISBN Verifier",
+        "uuid": "84645a75-35f9-44ff-bb1b-c4d8237c43d7",
+        "practices": [],
+        "prerequisites": [],
+        "difficulty": 1
```
Comment (0xNeshi) [inline config.json:861]: Set difficulty to 4

File: exercises/practice/isbn-verifier/.meta/example.cairo [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+    // Step 2: Check length of valid characters.
+    if size != 10 {
+        return false;
+    }
+
+    // Step 3: Calculate the sum based on the formula.
+    let mut i = 0;
+    let mut sum: u32 = 0;
+    let mut result = true;
```
Comment (0xNeshi) [inline exercises/practice/isbn-verifier/.meta/example.cairo:14]: Given how this flag is used, maybe a clearer name would be `let valid = true` - the name conveys the variable's purpose

File: exercises/practice/isbn-verifier/.meta/example.cairo [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        }
+        i += 1;
+    };
+
+    // Step 4: Return true if valid (sum % 11 == 0).
+    if result == true {
+        sum % 11 == 0
+    } else {
+        result
+    }
```
Comment (0xNeshi) [inline exercises/practice/isbn-verifier/.meta/example.cairo:38]: You can make this more concise:

```rust
// Step 4: Return true if valid (sum % 11 == 0).
result && sum % 11 == 0
```

**Review by 0xNeshi — COMMENTED** [review]
Excellent work @Falilah, this is really some idiomatic, clear and clean Cairo code :rocket: 

Just a couple of smaller things to address, and we're good to merge!

File: exercises/practice/isbn-verifier/src/lib.cairo [hunk 5]
```diff
@@ -0,0 +1,3 @@
+pub fn is_valid(isbn: ByteArray) -> bool {
+    panic!("implement `is_valid_isbn`")
```
Comment (0xNeshi) [inline exercises/practice/isbn-verifier/src/lib.cairo:2]:
```suggestion
    panic!("implement `is_valid`")
```

**Review by 0xNeshi — APPROVED** [review]
Loving this :heart_on_fire:

**Comment by Falilah** [comment]
> Loving this ❤️‍🔥

thank you very much 😇

**Pull request merged**